python/lsst/ts/wep/task/calcZernikesNeuralTask.py

```diff
@@ -0,0 +1,1964 @@
+        doc="List of Noll indices to calculate. Default is Z4-Z23 (4-23), "
+            "excluding piston (Z1), tip (Z2), and tilt (Z3) which are "
+            "typically not measured in wavefront sensing.",
+        dtype=int,
+        default=[4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23]
+    )
+    cropSize: pexConfig.Field = pexConfig.Field(
+        doc="Size of donut crop in pixels (width and height). Default is 160 pixels, "
+            "which matches the TARTS neural network training data format.",
+        dtype=int,
+        default=160
+    )
+
+
+class CalcZernikesNeuralTask(pipeBase.PipelineTask):
+    """Neural network-based Zernike estimation task using TARTS.
+
+    This class uses the TARTS (Telescope Active Optics Real-Time System)
+    neural network models to estimate Zernike coefficients from pairs of
+    intra and extra-focal exposures. Each exposure contains donut stamps
+    from one focal position, and the task processes them separately to
+    estimate Zernike coefficients for each side of the focal plane.
+    """
+    ConfigClass = CalcZernikesNeuralTaskConfig
+    _DefaultName = "calcZernikesNeuralTask"
+
+    def __init__(self, **kwargs: Any) -> None:
+        """Initialize the neural network-based Zernike estimation task.
+
+        This method sets up the TARTS neural network system, configures the
+        device (CPU or CUDA), and initializes the Noll indices for Zernike
+        coefficients. The TARTS system is loaded with the specified model
+        weights and dataset parameters from the configuration.
+
+        Parameters
+        ----------
+        **kwargs : Any
+            Additional keyword arguments passed to the parent PipelineTask
+            class.
+
+        Notes
+        -----
+        The initialization process:
+        1. Calls parent class constructor with super().__init__(**kwargs)
+        2. Sets up Noll indices from configuration (default: Z4-Z23)
+        3. Loads TARTS neural network models with specified weights
+        4. Sets models to evaluation mode with .eval()
+        5. Extracts crop size from TARTS system
+        6. Configures device (CPU/CUDA) and moves models accordingly
+
+        The TARTS system includes three main components:
+        - Wavenet: Estimates Zernike coefficients from donut images
+        - Alignnet: Handles image alignment and preprocessing
+        - Aggregatornet: Combines results from multiple donuts
+
+        Noll indices are configurable via the config.nollIndices parameter.
+        The default excludes Z1-Z3 (piston, tip, tilt) as these are typically
+        not measured in wavefront sensing.
+        """
+        super().__init__(**kwargs)
+
+        # Type annotation for mypy to understand the config structure
+        self.config: CalcZernikesNeuralTaskConfig
+
+        # Initialize task logger consistent with other tasks
+        self.log = logging.getLogger(type(self).__name__)  # type: ignore
```

Comment:
Are you sure you need to do this? Looking at the other tasks, they call self.log, without having to intialize it.